### ISSUE TITLE
Upgrade bundled Polymer to 1.2

### ISSUE THREAD

**Issue by warpech** [Starcounter/Starcounter#3022]
@malx122 says he needs `getEffeciveChildNodes`

@tomalec needs it for https://github.com/Starcounter/Starcounter/issues/2922

**Comment by warpech** [comment]
When done, remember updating the changelog

**Comment by miyconst** [comment]
I have tested our applications and `dom-bind-notifier` with Polymer 1.2

Most of the features work, but there are still some issues:

1. `dom-bind-notifier` test fails on updating nested arrays: `[["Alice", "Tomek"], ["Tomek", "Alice"]]`.
2. Applications fail to remove array items (tested with product, barcodes, images), but same thing works fine in my test ShoppingCart application. This may be caused by big nesting level in StarcounterSamples applications.

@warpech should I update `/sys/` folder?

**Comment by miyconst** [comment]
I have investigated the second point, seems like `dom-bind-notifier` for some reason notifies Polymer about a splice two times.

**Comment by warpech** [comment]
@malx122 do you mind if we upgrade to Polymer 1.2 only after we manage to point 2 listed above? We don't want to introduce regression.

Perhaps for now, you can upgrade Polymer in your installation:

```
staradmin kill all
cd "C:\Program Files\Starcounter\ClientFiles"
bower upgrade polymer
```

**Comment by miyconst** [comment]
Okay. The issue is located.

In most of our apps we have Master -> Details concept which requires this structure:

```html
<template is="dom-bind">
    <starcounter-include partial="{{model.CurrentPage}}"></starcounter-include>
    <dom-bind-notifier observed-object="{{model}}" path="model" deep></dom-bind-notifier>
</template>
```

And the partial for `starcounter-include` has same structure. Both `dom-bind-notifier`, from the master and from the details page observe same array, and both of them notify Polymer about the change in the array.

Polymer 1.1.x somehow ignores it and lets it go through, but Polymer 1.2.x tries to apply same splice second time and fails with an exception.

Later I will create a simple example to try to solve it, and now I have to get boarding.

**Comment by malx122** [comment]
Hm, can't do a manual upgrade since I will have the same problem then... Just solve the issue and I take it when its available :). (I do an ugly workaround for "geteffectiveChildNodes while waiting)

**Comment by miyconst** [comment]
I have located the issue. I believe it is Polymer's bug, but they may consider it as a "feature". Anyway, here is the issue I reported to Polymer: https://github.com/Polymer/polymer/issues/2680.

**Comment by warpech** [comment]
We decided to wait until Polymer responds to this issue

**Comment by tomalec** [comment]
Here is separated `dom-bind-notifier` upgrade https://github.com/Juicy/dom-bind-notifier/issues/8

**Comment by warpech** [comment]
Anything here useful for us? [Binding to Arrays -- Polycasts #35](https://www.youtube.com/watch?v=HBUb8frm2Tw)

**Comment by tomalec** [comment]
Not really, as I had to figure it out by myself a moth ago :/ 
But what is more important, I think we should not traverse through entire JSON tree to create observers for all arrays manually, and computed bindings does not help us either as we are usually using "dom-repeat", and even in that case we have problem with Polymer's API

**Comment by warpech** [comment]
Current status: Polymer 1.2 changes are available in https://github.com/StarcounterSamples/SysUpgrade repo that is being tested by @malx122. After he confirms it works, SysUpgrade can be merged to Starcounter `develop` and issues #3384, #3022, #2922 can be closed.

**Comment by warpech** [comment]
Upgrade to Polymer 1.3 done in #3384
